### PR TITLE
Feature/10834 show prüfpartner from allow list on consent ii screen

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Update the PrufPartener field in the second consent screen
PS: It takes a little time to show as we fetch the allowlist first

## Link to Jira
(https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10834)

## Screenshots
https://user-images.githubusercontent.com/15270737/144825770-1d965aee-73be-4197-8f6a-022362930811.mov

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/TicketValidation/TicketValidationCoordinator.swift [hunk 1]
```diff
@@ -131,11 +134,11 @@ final class TicketValidationCoordinator {
 	) {
 		let secondConsentViewController = SecondTicketValidationConsentViewController(
 			viewModel: SecondTicketValidationConsentViewModel(
-				// EXPOSUREAPP-10834: Needs to be set from allowlist when it's available: allowlist[0].serviceProvider
 				serviceIdentity: "",
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/TicketValidation/TicketValidationCoordinator.swift:137]: My expectation was that we just need to do what's written in the comment: pass in `ticketValidation.allowList.validationServiceAllowList.first?.serviceProvider ?? ""`. You need to make the `allowList` part of the protocol, but that's it. The allowList is already fetched and available here, isn't it?

File: src/xcode/ENA/ENA/Source/Scenes/TicketValidation/TicketValidationCoordinator.swift [hunk 2]
```diff
@@ -131,11 +134,11 @@ final class TicketValidationCoordinator {
...
-				serviceProvider: ticketValidation.initializationData.serviceProvider,
+				serviceProvider: "",
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Scenes/TicketValidation/TicketValidationCoordinator.swift:138]: I think this need to stay as it was

**Review by nickguendling — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/TicketValidation/TicketValidationCoordinator.swift (hunk 1 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Scenes/TicketValidation/TicketValidationCoordinator.swift:137]: no it is fetched inside the TicketValidation not the coordinator

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/MockTicketValidation.swift [hunk 3]
```diff
@@ -17,7 +17,7 @@ final class MockTicketValidation: TicketValidating {
 	}
 
 	let initializationData: TicketValidationInitializationData
-
+	var allowList = TicketValidationAllowList(validationServiceAllowList: [], serviceProviderAllowList: [])
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/MockTicketValidation.swift:20]:
```suggestion
	var allowList = TicketValidationAllowList(validationServiceAllowList: [], serviceProviderAllowList: [])
	
```

Would be nice to have a newline before the func :)

Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/MockTicketValidation.swift:20]:
```suggestion
	var allowList = TicketValidationAllowList(validationServiceAllowList: [], serviceProviderAllowList: [])
	
```

Would be nice to have a newline before the func :)

**Review by nickguendling — APPROVED** [review]

**Review by pascal-brause — APPROVED** [review]

**Pull request merged**